bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/handler/EnOceanBaseThingHandler.java

```diff
@@ -160,77 +173,27 @@ protected void updateChannels(EEPType eep, boolean removeUnsupportedChannels) {
         }
     }
 
-    protected HashSet<Channel> getLinkedChannels() {
-        if (linkedChannels != null) {
-            return linkedChannels;
-        }
-
-        linkedChannels = new HashSet<>();
-        if (channelState == null) {
-            channelState = new Hashtable<>();
-            lastEvents = new Hashtable<>();
-        }
-
-        for (Channel c : this.getThing().getChannels()) {
-            String id = c.getUID().getId();
-
-            if (isLinked(id)) {
-                {
-                    linkedChannels.add(c);
-                    channelState.putIfAbsent(id, UnDefType.UNDEF);
-                }
-            } else if (c.getKind() == ChannelKind.TRIGGER) {
-                linkedChannels.add(c);
-                lastEvents.putIfAbsent(id, "");
-            }
-        }
-
-        return linkedChannels;
-    }
-
     @Override
-    public void channelLinked(ChannelUID channelUID) {
-        super.channelLinked(channelUID);
-
-        String id = channelUID.getId();
-        if (linkedChannels == null) {
-            return;
-        }
-
-        linkedChannels.add(thing.getChannel(id));
-        channelState.putIfAbsent(id, UnDefType.UNDEF);
-        lastEvents.putIfAbsent(id, "");
+    public void thingUpdated(Thing thing) {
```

Comment:
Why do we override a method that only calls the super method?

Reply:
Yes you are right, I removed to less code. This unneccessary override can now be removed too. Thanks for the hint. Will fix it later.